dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/srt/SRTServletRequest.java

```diff
@@ -207,13 +216,23 @@ public class SRTServletRequest implements HttpServletRequest, IExtendedRequest,
     public SRTServletRequest(SRTConnectionContext context)
     {
         this._connContext = context;
-        this._requestContext = new SRTRequestContext(this);
+        this._requestContext = createRequestContext();
         this._in = createInputStream();
         if (TraceComponent.isAnyTracingEnabled()&&logger.isLoggable (Level.FINE)) {  //306998.15
             logger.logp(Level.FINE, CLASS_NAME,"SRTServletRequest", "this->"+this+": " + "inputStream is of type --> " + this._in);
         }
+        initRequestThreadData();
+    }
+
+    protected SRTRequestContext createRequestContext() {
+        return new SRTRequestContext(this);
     }
 
+    protected void initRequestThreadData() {
+        this.threadData = SRTServletRequestThreadData.getInstance();
+        this.threadData.init(null);
```

Comment:
```suggestion
        SRTServletRequestThreadData localThreadData = SRTServletRequestThreadData.getInstance();
        localThreadData.init(null);
        threadData = localThreadData;
```
This avoids the concurrency problem.

Reply:
Alternatively you could do:

```suggestion
        SRTServletRequestThreadData localThreadData = verifyThreadData();
        localThreadData.init(null);
```